### PR TITLE
fix(cli): restore objects with I prefix fails

### LINKED ISSUE #2945: Unable to restore a snapshot of file
I have a snapshot of file like following:

```
chetan@c1-chetan-prod:/media/data/chetan/backup/firebase-json/chetan-firebase.json
  2023-04-13 00:02:27 UTC Ix7836d96d0c93cd3a041c4e971a00629d 36.9 MB -rw-rw-r-- (daily-3)
  2023-04-14 00:04:23 UTC Ix466a865e9d87d2e56df371aedd32d649 37 MB -rw-rw-r-- (daily-2)
  2023-04-15 00:04:48 UTC Ix53cc4d4d982765c4009a638a923a76ca 37 MB -rw-rw-r-- (daily-1)
```

When I restore it, it throws error:

```bash
ubuntu@local-b1:~$ kopia restore Ix53cc4d4d982765c4009a638a923a76ca /home/ubuntu/firebase.json
Restoring to local filesystem (/home/ubuntu/firebase.json) with parallelism=8...
ERROR a snapshot time is needed to use a path as source
```

Providing `--snapshot-time` fails as well.

```bash
ubuntu@local-b1:~$ kopia restore Ix53cc4d4d982765c4009a638a923a76ca /home/ubuntu/firebase.json --snapshot-time="2023-04-15 00:00:03 UTC"
Restoring to local filesystem (/home/ubuntu/firebase.json) with parallelism=8...
ERROR no snapshots contain data for Ix53cc4d4d982765c4009a638a923a76ca
```

Kopia version on both is:

```bash
ubuntu@local-b1:~$ kopia --version
0.12.1 build: 5227d74996b6520f9f96e4203cfe00b832a60d5f from: kopia/kopia
```

This used to work earlier as my existing scripts started failing now. I'm not sure since when it might be broken.

For now, I have started creating snapshot of its parent directory as workaround.

### PR DESCRIPTION
**Pull request by NickIAm**
Fixes  #2945
The tryToConvertToPath would return an error when given an ID with an I prefix. Call object.parseID instead of index.parseID Also set the default for snapshot-time as that was missing. In my testing this seems to fix the issue

### CODE REVIEW

**Review by kakaroto — APPROVED** [review]
I don't know if my PR approval means anything, but this looks good and matches my suggested fix from #2945

**Comment by NickIAm** [comment]
Any chance of this getting merged? @jkowalski

**Comment by kakaroto** [comment]
@jkowalski I don't want to sound ungrateful for the work you're doing on this FOSS project, but you have a pretty major regression that is affecting many people, and we're stuck on 0.11.3 because of it, and you have this simple one-liner PR that fixes the regression and you just made a release but still without merging this one.

I understand that you may be busy with life, but can we get an ETA on when you could get this merged and a new release published please? It's been over a year since the regression was introduced and it prevents restoring snapshots, which is pretty critical IMHO.

Thank you.

**Comment by jkowalski** [comment]
the code is currently failing a test, so that needs to be fixed before we can merge. We should add a regression test to ensure restore with `I` prefix is tested on a regular basis. Please run `make` to make sure tests pass.

**Comment by kakaroto** [comment]
> the code is currently failing a test, so that needs to be fixed before we can merge. We should add a regression test to ensure restore with `I` prefix is tested on a regular basis. Please run `make` to make sure tests pass.

Thank you for responding to the PR!

It looks like the failure in unit test is "unexpected success" on `TestRestoreByPathWithoutTarget`.
I didn't try `make` locally, just checked the logs from github actions. I'm assuming it's because of the addition of `.Default("latest")` so that restoring by path can work without specifying a target, since the argument's help string says "default is latest" but the default wasn't actually set.

I'd say that (if my assumption on the failure is correct), the decision is yours:
- Should the unit test be fixed because a restore by path without target is now expected to work (and use latest as expected from the doc)?; or
- Should the default be removed and the argument help string modified to state that it's a required argument when restoring by path?

Thanks!

**Comment by NickIAm** [comment]
I've changed the unit test to reflect the --snapshot-time defaulting to "latest". It should now be passing. 

I'm not really familiar with how unit tests work in general as it's not something I've ever had to deal with in the past. I didn't even notice that test was failing before @kakaroto pointed it out. Now I know what to look for going forward.

Thanks for your patience while I am learning about how all this works

**Comment by kakaroto** [comment]
> I'm not really familiar with how unit tests work in general as it's not something I've ever had to deal with in the past. I didn't even notice that test was failing before @kakaroto pointed it out. Now I know what to look for going forward.
> 

Thanks for doing the changes! The unit tests hadn't failed yet until I believe @jkowalski approved the github action to run (see now they're also marked as "awaiting approval") and he's the one who noticed them failing too :)

**Comment by jkowalski** [comment]
Something is wrong with the branch. Please rebase against latest master and repush.

**Comment by NickIAm** [comment]
Sorry that got a bit messy. Looks better now

**Comment by kakaroto** [comment]
@jkowalski Hi, I just saw 0.16.0 and 0.16.1 releases and I'm wondering when this PR might get merged? It seems to keep falling through the cracks.
Bumping now so it can get a proper review/approval. 
Thanks

**Review by jkowalski — APPROVED** [review]

**Pull request merged**
